app/models/piece.rb

```diff
@@ -18,12 +17,47 @@ def move!(x, y)
     true
   end
 
+  # Return false if coordinates is taken but belongs to friendly piece
+  # otherwise return true
+  def capturable?(x, y)
+    return false unless position_taken?(x, y) && !friendly_piece?(x, y)
+    true
+  end
+
+  # if piece is capturable change piece's attributes
+  def capture!(x, y)
+    return false unless capturable?(x, y)
+    move!(x, y)
+    target_piece(x, y).update_attributes!(
+      x_position: nil,
+      y_position: nil,
+      captured: true
+    )
+  end
+
+  # Return true if target is a friendly piece
+  # otherwise return false
+  def friendly_piece?(x, y)
+    return true if color == target_piece(x, y).color
+    false
+  end
```

Comment:
This can be replaced by `color == target_piece(x, y).color`.  That statement itself will resolve to either true or false because of the logical comparison operator.  You don't need to explicitly return true or false.